lib/ethereumex/http_client.ex

```diff
@@ -4,7 +4,11 @@ defmodule Ethereumex.HttpClient do
   use Ethereumex.Client.BaseClient
   alias Ethereumex.Config
 
-  @spec post_request(binary(), []) :: {:ok | :error, any()}
+  @type empty_response :: :empty_response
+  @type invalid_json :: {:invalid_json, any}
+
+  @spec post_request(binary, []) ::
+          {:ok, any} | {:error, empty_response | invalid_json | any}
```

Comment:
can you please define `{:error, empty_response | invalid_json | any}` as a separate type and use it both for `post_request` and `decode_body`